<?xml version="1.0"?>
<launch>
    <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping_0" output="screen">
        <remap from="scan" to="/robot_0/base_scan" />

        <param name="map_frame" value="/robot_0/map" />
        <param name="base_frame" value="/robot_0/base_link" />
        <param name="odom_frame" value="/robot_0/odom" />
        
        <param name="use_tf_scan_transformation" value="true"/>
        <param name="use_tf_pose_start_estimate" value="false"/>
        <param name="pub_map_odom_transform" value="true"/>
        
        <param name="map_resolution" value="0.050"/>
        <param name="map_size" value="400"/>
        <param name="map_start_x" value="0.5"/>
        <param name="map_start_y" value="0.5" />
        <param name="map_multi_res_levels" value="2" />
        
        <param name="update_factor_free" value="0.4"/>
        <param name="update_factor_occupied" value="0.9" />    
        <param name="map_update_distance_thresh" value="0.4"/>
        <param name="map_update_angle_thresh" value="0.06" />

        <param name="advertise_map_service" value="true"/>
    </node>
    
    <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping_1" ns="robot_1" output="screen">
        <remap from="scan" to="/robot_1/base_scan" />

        <param name="map_frame" value="/robot_1/map" />
        <param name="base_frame" value="/robot_1/base_link" />
        <param name="odom_frame" value="/robot_1/odom" />
        
        <param name="use_tf_scan_transformation" value="true"/>
        <param name="use_tf_pose_start_estimate" value="false"/>
        <param name="pub_map_odom_transform" value="true"/>
        
        <param name="map_resolution" value="0.050"/>
        <param name="map_size" value="400"/>
        <param name="map_start_x" value="0.5"/>
        <param name="map_start_y" value="0.5" />
        <param name="map_multi_res_levels" value="2" />
        
        <param name="update_factor_free" value="0.4"/>
        <param name="update_factor_occupied" value="0.9" />    
        <param name="map_update_distance_thresh" value="0.4"/>
        <param name="map_update_angle_thresh" value="0.06" />
        
        <param name="pub_map_scanmatch_transform" value="false" />
        <param name="advertise_map_service" value="true"/>
    </node>
   
    <node name="relay_scan0" pkg="topic_tools" type="relay" args="/robot_0/base_scan /scan" />
    <node name="relay_scan1" pkg="topic_tools" type="relay" args="/robot_1/base_scan /robot_1/scan" />

    <node pkg="robot_detection" type="rerouter" name="rerouter" output="screen">
       <param name="robot_1_ns" type="string" value="/robot_1" />
       <param name="robot_0" type="string" value="/robot_0" />
       <param name="robot_1" type="string" value="/robot_1" />

       <param name="master" type="bool" value="true" />
       <param name="slave" type="bool" value="true" />
    </node>

    <node name="amcl" pkg="amcl_mod" type="amcl" output="screen">
       <remap from="/scan" to="/sync/scan" />
       <remap from="/map" to="/sync/map" />
       
       <param name="use_map_topic" type="bool" value="true" />
       <param name="first_map_only" type="bool" value="false" />
       <param name="reinitialize_filter" type="bool" value="false" />

       <param name="initial_cov_xx" type="double" value="10" />
       <param name="initial_cov_yy" type="double" value="10" />
       <param name="initial_cov_aa" type="double" value="10" />

       <param name="odom_model_type" type="string" value="diff" />
       <param name="laser_model_type" type="string" value="beam" />

       <param name="resample_interval" type="int" value="1" />
       
       <param name="odom_frame_id" type="string" value="/sync/odom" />
       <param name="base_frame_id" type="string" value="/sync/base_link" />
       <param name="global_frame_id" type="string" value="/sync/map" />
    </node>

    <!--<node name="tf_importer" type="tf_importer" pkg="robot_detection" />-->

    <!--<node pkg="rosservice" type="rosservice" name="global_loc" args="call - -wait /global_localization"/>-->
 
    <include file="$(env RABOT)/Config/Simulation/MultipleRobots/multiple_robots.launch" />
   
    <node pkg="joy" type="joy" name="joy">
        <param name="autorepeat_rate" value="15.0" />
    </node>

    <node pkg="teleop" type="teleop" name="teleop" output="screen" />

</launch>
